src/interop/js-interop/js-types.md

```diff
@@ -0,0 +1,189 @@
+---
+title: JS Types
+description: Usage information about the core types in JS interop.
+---
+
+## Motivation
+
+Dart values and JS values belong to separate domains. When compiling to Wasm,
+they belong in separate *runtimes* as well. As such, users should treat JS
+values as foreign types. In order to provide types for JS values, we use the set
+of JS types available in `dart:js_interop`. These are extension types prefixed
+with `JS` and are called "JS types".
+
+Importantly, these types have compiler-specific representation types. This means
+that their runtime type will differ based on whether code is compiled to Wasm or
+JS. In order to interact with and examine these JS values, we use `external`
+interop members.
```

Comment:
Probably want a link on both "representation types" and `external` for definitions, but like other links, that'll have to wait for more docs to form

Reply:
Done!